symfony/src/Service/PatientStatusImportService.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace App\Service;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Pmi\PatientStatus\PatientStatus as PmiPatientStatus;
+use Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface;
+use Symfony\Component\Form\FormError;
+
+class PatientStatusImportService
+{
+    public function __construct(UserService $userService, EntityManagerInterface $em, ParameterBagInterface $params)
+    {
+        $this->userService = $userService;
+        $this->em = $em;
+        $this->params = $params;
+    }
+
+    public function extractCsvFileData($file, &$form, &$patientStatuses)
+    {
+        $fileHandle = fopen($file->getPathname(), 'r');
+        $headers = fgetcsv($fileHandle, 0, ",");
+        $validHeaders = ['participantid', 'status', 'comments'];
+        // Guess file format using headers
+        if (count($headers) !== 3) {
+            $form['patient_status_csv']->addError(new FormError("Invalid file format"));
+            return;
+        }
+        // Check column headers
+        foreach ($headers as $header) {
+            // Handle bom
+            $header = str_replace("\xEF\xBB\xBF", '', $header);
+            if (!in_array(str_replace(' ', '', strtolower($header)), $validHeaders)) {
+                $form['patient_status_csv']->addError(new FormError("Invalid column headers"));
+                return;
+            }
+        }
```

Comment:
I'm not sure we need to enforce specific strings in the header row. In the instructions, these three valid headers are never mentioned. I think we should just check that the first row returns something and continue.

Reply:
Sure!